source/common/upstream/health_checker_impl.cc

```diff
@@ -264,9 +264,12 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResponseComplete() {
     break;
   }
 
-  if ((response_headers_->Connection() &&
+  if ((response_headers_->Connection() && protocol_ != Http::Protocol::Http2 &&
```

Comment:
I was going to suggest a release note for this (as I assumed it changed the behavior for any H2 which was sending an H1 header) but according to the H2 spec the connection header is entirely illegal* and it turns out nghttp2 rejects it so this is a no-op.

https://http2.github.io/http2-spec/#rfc.section.8.1.2.2

I'd be inclined to leave this as-is without the Http::Protocol::Http2 check just on the theory we should keep the code simpler and not special case something illegal.